libs/core/utils/nameof.ts

```diff
@@ -0,0 +1,5 @@
+/**
+ * Get function for type that return its passed field name.
+ * Has compilation error if field doesn't exist.
+ */
+export const nameofFactory = <T>() => (fieldName: keyof T) => fieldName;
```

Comment:
For omit not object types

```suggestion
export const nameofFactory = <T extends Record<string, any>>() => (fieldName: keyof T) => fieldName;
```

Reply:
Linter warn me about `any` using. Can I make T extended object? It works for my case.

Reply:
Yes, in this case you can use `object`

Reply:
Just for you know (it not about your case, but about types), i suggested `Record<string, any>` because it analog object, but with ability to access fields

in `object` case we will get error

see [example](https://www.typescriptlang.org/play?#code/MYewdgzgLgBAhgLhgJQKagE4BMA80MCWYA5gDTxgCeAfDALwwDeMAUDDAGYGoA2WSAcgAWvHiAGkWAXxYs4AOi68sAblmhIsAEZIQWgFbpYDRm07c+gkTzEDpsrfOtigA)

Reply:
Got it